cli/src/test/java/com/devonfw/tools/ide/io/FileAccessImplTest.java

```diff
@@ -0,0 +1,299 @@
+    assertThat(dir.resolve("d1/d11/link2")).existsNoFollowLinks();
+    assertThat(dir.resolve("d1/d11/link3")).existsNoFollowLinks();
+    assertThat(dir.resolve("d1/d11/link4")).existsNoFollowLinks();
+    assertThat(dir.resolve("d1/d11/link5")).existsNoFollowLinks();
+    assertThat(dir.resolve("d1/d11/link6")).existsNoFollowLinks();
+    assertThat(dir.resolve("d1/d11/link7")).existsNoFollowLinks();
+    assertThat(dir.resolve("d2/d22/link8")).existsNoFollowLinks();
+    assertThat(dir.resolve("d2/link9")).existsNoFollowLinks();
+    assertThat(dir.resolve("link10")).existsNoFollowLinks();
+  }
+
+  private void assertSymlinksAreBroken(Path dir, boolean readLinks) throws IOException {
+
+    assertSymlinkIsBroken(dir.resolve("d1/d11/link1"), readLinks);
+    assertSymlinkIsBroken(dir.resolve("d1/d11/link2"), readLinks);
+    assertSymlinkIsBroken(dir.resolve("d1/d11/link3"), readLinks);
+    assertSymlinkIsBroken(dir.resolve("d1/d11/link4"), readLinks);
+    assertSymlinkIsBroken(dir.resolve("d1/d11/link5"), readLinks);
+    assertSymlinkIsBroken(dir.resolve("d1/d11/link6"), readLinks);
+    assertSymlinkIsBroken(dir.resolve("d1/d11/link7"), readLinks);
+    assertSymlinkIsBroken(dir.resolve("d2/d22/link8"), readLinks);
+    assertSymlinkIsBroken(dir.resolve("d2/link9"), readLinks);
+    assertSymlinkIsBroken(dir.resolve("link10"), readLinks);
+  }
+
+  private void assertSymlinkIsBroken(Path link, boolean readLinks) throws IOException {
+
+    try {
+      Path realPath = link.toRealPath();
+      if (Files.exists(realPath)) {
+        fail("The link target " + realPath + " (from toRealPath) should not exist");
+      }
+    } catch (IOException e) { // toRealPath() throws exception for junctions
+      assertThat(e).isInstanceOf(NoSuchFileException.class);
+    }
+    if (readLinks) {
+      Path readPath = Files.readSymbolicLink(link);
+      if (Files.exists(readPath)) {
+        fail("The link target " + readPath + " (from readSymbolicLink) should not exist");
+      }
+    }
+  }
+
+  /*
+   * only pass readLinks = true when junctions are not used.
+   */
+  private void assertSymlinksWork(Path dir, boolean readLinks) {
+
+    assertSymlinkToRealPath(dir.resolve("d1/d11/link1"), dir.resolve("d1"));
+    assertSymlinkToRealPath(dir.resolve("d1/d11/link2"), dir.resolve("d1/d11"));
+    assertSymlinkToRealPath(dir.resolve("d1/d11/link3"), dir.resolve("d1/d11/d111"));
+    assertSymlinkToRealPath(dir.resolve("d1/d11/link4"), dir.resolve("d1/d11/d111/d1111"));
+    assertSymlinkToRealPath(dir.resolve("d1/d11/link5"), dir.resolve("d2"));
+    assertSymlinkToRealPath(dir.resolve("d1/d11/link6"), dir.resolve("d2/d22"));
+    assertSymlinkToRealPath(dir.resolve("d1/d11/link7"), dir.resolve("d2/d22/d222"));
+    assertSymlinkToRealPath(dir.resolve("d2/d22/link8"), dir.resolve("d1"));
+    assertSymlinkToRealPath(dir.resolve("d2/link9"), dir.resolve("d1"));
+    assertSymlinkToRealPath(dir.resolve("link10"), dir.resolve("d1"));
```

Comment:
I often see such pattern in JUnits ("lack of business oriented test-data/-scenario") - however it is very hard to read and understand such "technical random setup".
I figured it corresponds to `createSymlinks` but if you use more speaking names and an example that somehow makes sense it is much easier for others to read the code. Example:
Create symlink from `long-path/that/should-have/some-shortcut` to `shortcut` and users can much better understand your expected result than why `d1/d11/link4` points to `d1/d11/d111/d1111`.

Please note: This is only for your information to think about for the next time. IMHO your test is OK and you do not have to refactor this. I am already happy that you write a test and cover so many cases what is great. Just always trying to give feedback so you can learn and still improve...

Reply:
Thanks! I tried to make the link names a bit more clear. I left the names of the dirs the same, because I dont want to have super long path names.